src/horust/supervisor/process_spawner.rs

```diff
@@ -173,6 +202,43 @@ fn redirect_output(
     Ok(())
 }
 
+fn open_next_chunk(base_path: &Path) -> io::Result<File> {
+    let mut count = 1;
+    let mut path = base_path.to_path_buf();
+
+    while path.is_file() {
+        // let path_str = format!("{}.{count}", base_path,  path.to_string_lossy().to_string() + format!(".{count}").as_ref();
+        let path_str = format!("{}.{count}", base_path.to_string_lossy());
+        // path = Path::new(&path_str).to_path_buf();
+        path = Path::new(&path_str).to_path_buf();
```

Comment:
I think PathBuff::from(path_str) would also work

Reply:
Rewrote the loop to use Path instead, as suggested below.